cmd/terramate/e2etests/general_test.go

```diff
@@ -312,13 +312,30 @@ func TestFailsOnChangeDetectionIfCurrentBranchIsMainAndItIsOutdated(t *testing.T
 
 	assertRunResult(t, ts.listChangedStacks(), wantRes)
 
-	assertRunResult(t, ts.run(
-		"run",
-		"--changed",
-		testHelperBin,
-		"cat",
-		mainTfFile.HostPath(),
-	), wantRes)
+	testrun := func() {
+		assertRunResult(t, ts.run(
+			"run",
+			testHelperBin,
+			"cat",
+			mainTfFile.HostPath(),
+		), wantRes)
+
+		assertRunResult(t, ts.run(
+			"run",
+			"--changed",
+			testHelperBin,
+			"cat",
+			mainTfFile.HostPath(),
+		), wantRes)
+	}
+
+	testrun()
+
+	git.CheckoutNew("branch")
+	stack.CreateFile("tempfile2", "any content")
+	git.CommitAll("add tempfile2")
+
+	testrun()
```

Comment:
hmm so now we always check with and without --changed + we are also checking the detection if you are not on an outdated main, but on an outdated branch.

Reply:
yeah but the branch you are is not important anymore, we only check if local origin/main is updated.

Reply:
thanks for clarifying :heart: